### PR TITLE
HPCC-17075 Configgen xslt warnings handled differently than exceptions

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@ghalliday Can you fetch this branch and test that it fixes the issues you were having with running the regression suite?

**Comment by ghalliday** [comment]
@Michael-Gardner yes that fixes the problem - I can now start my environment without patching the files.

If I run "configgen -validateonly -env /etc/HPCCSystems/environment.xml" I get the following output:

```
00000000 2017-03-21 16:09:55.497  8024  8024 "WARNING: Cannot create /tmp/82129795/myeclserver/eclserver.xml
using XSL transform /opt/HPCCSystems/componentfiles/configxml/eclserver.xsl

[5657: MySQL server is not specified for ECL server!] "
00000001 2017-03-21 16:11:27.046  8045  8045 "WARNING: Couldn't delete file @temp/EclWatch.xml+EclWatch"
00000002 2017-03-21 16:11:27.046  8045  8045 "WARNING: Couldn't delete file @temp/ws_ecl2.xml+ws_ecl"
```
The complaint that it cannot create eclserver.xml is not correct, but I would merge as is since it is better than before.

File: deployment/deploy/DeployTask.cpp [hunk 1]
```diff
@@ -492,6 +476,28 @@ class CDeployTask : public CInterface, implements IDeployTask
      return (m_errorCode == 0);
    }
    //---------------------------------------------------------------------------
+   //  messageFormat
+   //---------------------------------------------------------------------------
+   void formatMessage(const char * target, const char * xsl, StringBuffer & in)
+   {
+	   StringBuffer message;
```
Comment (ghalliday) [inline deployment/deploy/DeployTask.cpp]: whitespace: Strange indentation.

File: deployment/deploy/DeployTask.cpp [hunk 2]
```diff
@@ -492,6 +476,28 @@ class CDeployTask : public CInterface, implements IDeployTask
...
+         message.remove(0, len);
+
+       //remove the excessive info about XSLT context when this was thrown
+       const char* begin = message.str();
+       const char* end   = strstr(begin, ", style tree node:");
+
+       if (end)
+         message.setLength(end-begin);
+
+       m_errorString.appendf("Cannot create %s\nusing XSL transform %s\n\n%s", target, xsl, message.str());
```
Comment (ghalliday) [inline deployment/deploy/DeployTask.cpp]: Should be conditional on whether it is a warning or an error.

**Comment by Michael-Gardner** [comment]
@ghalliday Got rid of the indentation issues.  Also made the formatMessage only output a failure to write the final file if it's an error and not a warning.  So you should see a message output just from the warning log now.  It tells where the warning came from (xslt file) and the warning message from that file.

File: deployment/deploy/DeployTask.cpp [hunk 3]
```diff
@@ -492,6 +476,31 @@ class CDeployTask : public CInterface, implements IDeployTask
...
    }
    //---------------------------------------------------------------------------
+   //  messageFormat
+   //---------------------------------------------------------------------------
+   void formatMessage(const char * target, const char * xsl, StringBuffer & in, bool isError=false)
+   {
+     StringBuffer message;
+     message.set(in.str());
+     //for better readability of warning, remove redundant prefix of "[XSLT warning: ", if present
+       const char* pattern = "[XSLT warning: ";
```
Comment (ghalliday) [inline deployment/deploy/DeployTask.cpp]: trivial indent: rest of the function is indented 2 spaces extra.

**Comment by ghalliday** [comment]
@Michael-Gardner happy to merge - do you want to fix the indentation first?

**Comment by Michael-Gardner** [comment]
@ghalliday Fixed the indentation.  Ready for a final look.

**Pull request merged**